mindsdb/integrations/handlers/azure_blob_handler/azure_blob_handler.py

```diff
@@ -0,0 +1,364 @@
+        data = []
+        for path in tables:
+            path = path.replace('`', '')
+            item = {
+                'path': path,
+                'name': path[path.rfind('/') + 1:],
+                'extension': path[path.rfind('.') + 1:]
+            }
+
+            data.append(item)
+
+        return pd.DataFrame(data=data, columns=self.get_columns())
+
+    def get_columns(self) -> List[str]:
+        return ["path", "name", "extension", "content"]
+
+
+class FileTable(APIResource):
+
+    def list(self, targets: List[str] = None, table_name=None, *args, **kwargs) -> pd.DataFrame:
+        return self.handler.read_as_table(table_name)
+
+    def add(self, data, table_name=None):
+        df = pd.DataFrame(data)
+        return self.handler.add_data_to_table(table_name, df)
+
+
+class AzureBlobHandler(APIHandler):
+    """
+    This handler handles connection and execution of the SQL statements on Azure Blob.
+    """
+
+    name = 'azureblob'
+    supported_file_formats = ['csv', 'tsv', 'json', 'parquet']
+
+    def __init__(self, name: str, **kwargs):
+        super().__init__(name)
+        """ constructor
+        Args:
+            name (str): the handler name
+        """
+
+        self.connection = None
+        self.is_connected = False
+        self._tables = {}
+        self.storage_account_name = None
+        self.account_access_key = None
+        self._files_table = ListFilesTable(self)
+        self.container_name = None
+
+        connection_data = kwargs.get('connection_data')
+        if 'storage_account_name' in connection_data:
+            self.storage_account_name = connection_data['storage_account_name']
+
+        if 'account_access_key' in connection_data:
+            self.account_access_key = connection_data['account_access_key']
+
+        if 'container_name' in connection_data:
+            self.container_name = connection_data['container_name']
+
+        if 'connection_string' in connection_data:
+            self.connection_string = connection_data['connection_string']
+
+    def connect(self) -> BlobServiceClient:
+        """ Set up any connections required by the handler
+        Should return output of check_connection() method after attempting
+        connection. Should switch self.is_connected.
+        Returns:
+            HandlerStatusResponse
+        """
+        if self.is_connected is True:
+            return self.connection
+
+        sas_token = generate_account_sas(
+            account_name=self.storage_account_name,
+            account_key=self.account_access_key,
+            resource_types=ResourceTypes(service=True, container=True, object=True),
+            permission=AccountSasPermissions(read=True, list=True),
+            expiry=datetime.now() + timedelta(hours=1)
+        )
+
+        blob_service_client = BlobServiceClient(
+            account_url=f"https://{self.storage_account_name}.blob.core.windows.net",
+            credential=sas_token
+        )
```

Comment:
I think we need to improve how the authentication works here; most Azure users are probably going to be unwilling to provide the access keys to the storage account because they allow full access. They would be more willing to pass in a generated SAS instead. Therefore, while keeping access keys as an option, I think we should also allow SASs as a connection parameter.

Further, I don't think it is necessary to generate a SAS when using connection string/access key authentication. We can simply do `BlobServiceClient.from_connection_string(connection_string)`.

So, I think there are essentially 2 (+1 optional) ways for users to provide their authentication credentials:
1. Shared Access Signature (`shared_access_signature`): this should be the recommended method because the credential can be generated with only the required access. However, we should be very clear about the access that needs to be provided in the documentation (README).
2. Connection String (`connection_string`): this parameter is already available, you just need to change how it is used in the code. Plus, it would be best if we can just mention where the connection string can be retrieved from in the README.
3. Account Name + Access Key (`storage_account_name` + `access_key`): this method is optional because it is essentially the same as 2, just in parts. I don't think it is really necessary since the connection string is readily available to be copied.

Reply:
connection method, & readme updated

Reply:
I don't see `shared_access_signature` added as a connection parameter. Are you only looking to support authentication options 1 and 2 from above? If so, we don't need to include `storage_account_name` as a parameter either, right?

Reply:
I use connection_string from different source. 
In azure, we can find `connection string` under Access keys menu & Shared Access Signature menu.
The new used `connection string` come from Shared Access Signature menu which I expect has more limited & controlled access.

Ah! You are correct Minura! I almost miss that `storage_account_name`, thanks for reminding me

Reply:
As far as I know (you will need to test this), the syntax for connecting via a SAS URL and a connection string (from the menu for Access Keys) is different. The syntax used for SAS URLs look like this:
`BlobServiceClient(account_url=sas_url`

So, we can have a new parameter called `sas_url` and enable authentication through that.

Reply:
And other consideration that led me to use `connection string` is duckdb. Currently, my successful attempt to make duckdb connect to azure only by using `connection string`... but I believe there's an alternative way

Reply:
just read your new comment after posted my last one

Reply:
Yes the provided params are different if we use SAS url & connection string
Would u consider use SAS token instead SAS url? Azure seperate the SAS url for each its services.. (blue lines)
![Screenshot 2024-10-30 at 19-20-16 jevonstoragecreate - Microsoft Azure](https://github.com/user-attachments/assets/481d67a9-d9c3-4d9c-a709-d74ed7c296a5)

But if we use SAS token, we might not need to provide new param in future because SAS token / connection string already take care of them

Reply:
@Tryxns OK, since this is causing confusion, let's just enable authentication via a connection string for now. We can tackle other methods later.

Reply:
Applied